packages/teams-js/src/public/appWindow.ts

```diff
@@ -85,39 +61,21 @@ export class ParentAppWindow implements IAppWindow {
 
   /**
    * Send a message to the AppWindow.
+   *
    * @param message - The message to send
-   * @returns Promise that will be fulfilled when the AppWindow posts back a response
-   */
-  public postMessage(message: any): Promise<void>;
-  /**
-   * @deprecated
-   * As of 2.0.0-beta.1, please use {@link ParentAppWindow.postMessage ParentAppWindow.postMessage(message: any): Promise\<void\>} instead.
-   * @param message - The message to send
-   * @param onComplete - The deprecated way of invoking a callback to know if the postMessage has been success/failed.
+   * @param onComplete - The callback to know if the postMessage has been success/failed.
    */
-  public postMessage(message: any, onComplete: (status: boolean, reason?: string) => void): void;
-  public postMessage(message: any, onComplete?: (status: boolean, reason?: string) => void): Promise<void> {
+  public postMessage(message: any, onComplete?: (status: boolean, reason?: string) => void): void {
     ensureInitialized(FrameContexts.task);
-    return this.postMessageHelper(message)
-      .then(() => {
-        if (onComplete) {
-          onComplete(true);
-        }
-      })
-      .catch((err: Error) => {
-        if (onComplete) {
-          onComplete(false, err.message);
-          return;
-        }
-        throw err;
-      });
-  }
-  public postMessageHelper(message: any): Promise<void> {
-    return new Promise<void>(resolve => {
-      resolve(send('messageForParent', message));
-    });
+    sendMessageToParent('messageForParent', [message], onComplete ? onComplete : getGenericOnCompleteHandler());
   }
 
+  /**
+   * Add a listener that will be called when an event is received from this AppWindow.
```

Comment:
Would it be more correct to say "Add a listener that will be called when an event is received from the ParentAppWindow"?